### PR TITLE
Feat/verification provider + range proofs on wallet-sdk-cli

### PR DESCRIPTION
**Pull request by maycon-mello**
- The current verification flow was partially implemented in the wallet-sdk and some logic was placed in the wallet-app. We had a lot of hooks to, which were hard to test. I've moved all its logic to the verificationProvider, so that we have a single place to implement the verification logic and it will be better to maintain

### CODE REVIEW

File: packages/cli/src/commands/messages.ts [hunk 1]
```diff
@@ -12,44 +14,71 @@ messagesCommand
...
     });
 
-    // send relay service message
-    // if multiple dids, ask with did to use
+    console.log('Message sent!');
+    console.log(result);
+  });
+
+const DIDAddressBook = {
+  iosEmulator: 'did:key:z6MkuZRbStWnxNunsGRbWBfMdZaE9iqrgFkYZXi1C6dMtREi',
```
Comment (mike-parkhill) [inline packages/cli/src/commands/messages.ts:55]: should these be read in from a config or env file?

File: packages/cli/src/commands/verification.ts [hunk 2]
```diff
@@ -1,28 +1,117 @@
...
-import axios from 'axios';
+import select, {Separator} from '@inquirer/select';
+import checkbox from '@inquirer/checkbox';
+import input from '@inquirer/input';
 import {IWallet} from '@docknetwork/wallet-sdk-core/lib/types';
 import {getWallet} from '../helpers';
-
-const verificationCommands = new Command('dids');
+import {createVerificationController} from '@docknetwork/wallet-sdk-core/src/verification-controller';
+// import template from '@docknetwork/wallet-sdk-core/src/fixtures/iiw-template.json';
```
Comment (mike-parkhill) [inline packages/cli/src/commands/verification.ts:8]: is this commented import needed?

File: packages/core/src/credential-provider.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export function createCredentialProvider({wallet}: {wallet: IWallet}) {
+  function getCredentials(type: string = 'VerifiableCredential') {
+    return wallet.getDocumentsByType(type);
+  }
+
+  return {
+    getCredentials,
+    // TODO: move credential validity check to this provider
```
Comment (mike-parkhill) [inline packages/core/src/credential-provider.ts:14]: are these still in the app codebase?

File: packages/wasm/src/services/credential/service.ts [hunk 4]
```diff
@@ -167,6 +176,8 @@ class CredentialService {
         }
       });
       credentialJSON['@context'].push(customContext);
+
+      console.log();
```
Comment (mike-parkhill) [inline packages/wasm/src/services/credential/service.ts:180]: ?

Comment (maycon-mello) [reply packages/wasm/src/services/credential/service.ts]: I've removed it

File: packages/cli/src/commands/verification.ts (hunk 2 above)
Comment (maycon-mello) [reply packages/cli/src/commands/verification.ts]: I've removed it

File: packages/core/src/credential-provider.ts (hunk 3 above)
Comment (maycon-mello) [reply packages/core/src/credential-provider.ts:14]: Yes, still need to move this part

File: packages/cli/src/helpers.ts [hunk 5]
```diff
@@ -25,3 +32,29 @@ export async function getWallet(): Promise<IWallet> {
 export function getDIDProvider(): IDIDProvider {
   return didProvider;
 }
+
+export function getCredentialProvider(): ICredentialProvider {
+  return credentialProvider;
+}
+
+export async function selectCredential() {
+  const credentials = await credentialProvider.getCredentials();
```
Comment (mike-parkhill) [inline packages/cli/src/helpers.ts:41]: would be cool if the credentialProvider could just iterate over a folder so I could throw different ones easily as I'm testing

**Review by mike-parkhill — APPROVED** [review]

File: packages/react-native/bundler/index.js [hunk 6]
```diff
@@ -44,13 +45,34 @@ const getWebpackConfig = ({entry, path, filename}) => ({
   },
   module: {
     rules: [
+      // {
```
Comment (mike-parkhill) [inline packages/react-native/bundler/index.js:48]: are these commented sections coming back?

**Review by mike-parkhill — APPROVED** [review]

**Pull request merged**